### PR TITLE
adding state to resource

### PR DESCRIPTION
**Pull request by truhacevkir**
- Adding state to Resource which will be used to populate log message.
- Minor clean up in resource service - more clean up to do.

### CODE REVIEW

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java [hunk 1]
```diff
@@ -147,26 +148,6 @@ public Publication createPublication(UserInstance userInstance, Publication inpu
...
-        toResource.setStatus(status);
-    }
-
-    public Publication createPublicationWithPredefinedCreationDate(Publication inputData) {
-        Resource newResource = Resource.fromPublication(inputData);
-        newResource.setIdentifier(identifierSupplier.get());
-        newResource.setCreatedDate(inputData.getCreatedDate());
-        return insertResource(newResource);
-    }
-
```
Comment (truhacevkir) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:169]: Have been used in tests only.

File: publication-commons/src/test/java/no/unit/nva/publication/service/impl/MigrationTests.java [hunk 2]
```diff
@@ -183,57 +184,6 @@ void shouldWorkWithEmptyEntityDescription() {
         assertThat(resource.getCuratingInstitutions(), hasSize(0));
     }
 
-    @Deprecated
```
Comment (truhacevkir) [inline publication-commons/src/test/java/no/unit/nva/publication/service/impl/MigrationTests.java:186]: Removing migration tests

File: publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java [hunk 3]
```diff
@@ -196,23 +195,6 @@ void shouldInstantiateResourceServiceForProvidedTable() {
         assertThat(tableNames, hasItem(customTable));
     }
 
-    @Test
```
Comment (truhacevkir) [inline publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java:199]: Removed test and method in resource service it was testing - method is never used in production.

File: publication-commons/src/main/java/no/unit/nva/publication/model/business/publicationstate/State.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import no.unit.nva.publication.model.business.User;
+
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+@JsonSubTypes({
+    @JsonSubTypes.Type(CreatedState.class),
+    @JsonSubTypes.Type(PublishedState.class),
+    @JsonSubTypes.Type(UnpublishedState.class),
+    @JsonSubTypes.Type(DeletedState.class)
+})
+public interface State {
```
Comment (torbjokv) [inline publication-commons/src/main/java/no/unit/nva/publication/model/business/publicationstate/State.java:16]: I suggest ResourceLogEntry or something like that?

File: publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java [hunk 5]
```diff
@@ -482,7 +482,7 @@ public ResourceBuilder copy() {
                    .withRightsHolder(getRightsHolder())
                    .withCuratingInstitutions(getCuratingInstitutions())
                    .withImportDetails(getImportDetails())
-                   .withState(getState());
+                   .withResourceEvent(getResourceEvent());
```
Comment (torbjokv) [inline publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java:485]: will this end up in search?

**Review by torbjokv — APPROVED** [review]

File: publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java (hunk 5 above)
Comment (truhacevkir) [reply publication-commons/src/main/java/no/unit/nva/publication/model/business/Resource.java:485]: I dont think so, we will see :)

**Pull request merged**